contrib/nydusd-http-server/src/main.rs

```diff
@@ -0,0 +1,236 @@
+                self.start,
+                self.end,
+                self.end - self.start
+            );
+            let size = self.end - self.start + 1;
+            let mut file = match fs::File::open(self.path) {
+                Ok(f) => f,
+                Err(_err) => {
+                    return Err(Status::NotFound);
+                }
+            };
+            match file.seek(io::SeekFrom::Start(self.start)) {
+                Ok(file) => file,
+                Err(_err) => {
+                    return Err(Status::InternalServerError);
+                }
+            };
+            Response::build()
+                .streamed_body(ReaderStream! {
+                const BUFSIZE : usize=1048576;  //bufsize:2M
+                let mut buf = [0; BUFSIZE];
+                let mut readed = 0;
+                    while readed < size
+                {
+                  let unread = (size - readed) as usize;
+                  match file.read(&mut buf)
+                  {
+                    Ok(n) =>
+                    {
+                     if n == 0
+                     {
+                      break
+                     }
+                     readed += n as u64;
+                     if unread < BUFSIZE {
+                         let part=&buf[0..unread];
+                         yield io::Cursor::new(part.to_vec());
+                     }
+                     else{
+                          yield io::Cursor::new(buf.to_vec());
+                     }
+                    }
+                    Err(_err) => break,
+                  }
+                }
+                  })
+                .raw_header("Docker-Content-Digest", self.dis)
+                .raw_header("Content-Type", "application/octet-stream")
+                .raw_header("Content-Range", rangestring)
+                .ok()
+        }
+    }
+
+    #[derive(Responder)]
+    pub enum StoredData {
+        AllFile(Option<CachedFile>),
+        Range(RangeStream),
+    }
+
+    #[get("/<_namespace>/<_repo>/blobs/<digest>")]
+    pub async fn fetch_blob(
+        _namespace: PathBuf,
+        _repo: PathBuf,
+        digest: String,
+        headerdata: HeaderData,
+    ) -> Result<StoredData, Status> {
+        let args: Vec<String> = env::args().collect();
+        let mut filepath = Path::new(&args[1]).join("blobs");
+        let dis = &digest[7..];
+        if filepath.is_dir() {
+            filepath.push(&dis);
+        }
+        if !filepath.is_file() {
+            return Err(Status::NotFound);
+        }
+        //if no range in Request header,return fetch blob response
+        if headerdata.range.is_empty() {
+            return Ok(StoredData::AllFile(
+                NamedFile::open(filepath)
+                    .await
+                    .ok()
+                    .map(|nf| CachedFile(nf, dis.to_string())),
+            ));
+        } else {
+            let metadata = match fs::metadata(&filepath) {
```

Comment:
This op should be cached, how about opening all blob files fd on the server booting.